charts/redis-ha/templates/redis-ha-statefulset.yaml

```diff
@@ -467,8 +467,8 @@ spec:
           - name: REDIS_PASSWORD
             valueFrom:
               secretKeyRef:
-              {{- if .Values.existingSecret }}
-                name: {{ .Values.existingSecret }}
+              {{- if tpl (.Values.existingSecret | default "" ) . }}
```

Comment:
Is the tpl replace in the conditional necessary?

Reply:
Well, the idea is that if the template resolves to the empty string, then it should be equivalent to the case, as if the existing secret is not provided. 

Assume, e.g., the case when you have redis as subchart:
```
global:
  secret:
    passwordsSecret: "" # Specify the user provided secret with password, if empty the chars will generate a random password
    
redis:
  existingSecret: '{{ .Values.global.secret.passwordsSecret }}'
```

In this case you have non empty `(redis).existingSecret` value with template, which however resolves to empty string.
 I believe in these cases you want to behave exactly as if the `existingSecret` is empty; that is why we should have `tpl` even in the conditions.
 
 This behavior is tested in Test 5 (at the bottom of) , in mine [comment](https://github.com/DandyDeveloper/charts/pull/253#issuecomment-1520329188) above.